cmd/admin-trace.go

```diff
@@ -378,11 +480,178 @@ func matchingOpts(ctx *cli.Context) (opts matchOpts) {
+		rotateTraceLogsFrequency, e = time.ParseDuration(ctx.String("rotate-trace-logs-frequency"))
+		fatalIf(probe.NewError(e).Trace(ctx.String("rotate-trace-logs-frequency")), "Unable to parse input trace logs frequency.")
+		// Enforce minimum frequency
+		if rotateTraceLogsFrequency < time.Minute {
+			fatalIf(errDummy().Trace(), "Frequency must be a minimum of every 1 minute.")
+		}
+		// Obtain time interval type. Default to day
+		pattern := regexp.MustCompile(`.*([a-zA-Z])+`)
+		subMatches := pattern.FindStringSubmatch(ctx.String("rotate-trace-logs-frequency"))
+		switch subMatches[1] {
+		case "m":
+			traceLogsIntervalType = time.Second * 60
+		case "h":
+			traceLogsIntervalType = time.Second * 60 * 60
+		case "d":
+			traceLogsIntervalType = time.Second * 60 * 60 * 24
+		default:
+			traceLogsIntervalType = time.Second * 60 * 60 * 24
+		}
+	}
+
+	if ctx.String("rotate-trace-logs-pattern") != "" {
+		rotateTraceLogsPattern = ctx.String("rotate-trace-logs-pattern")
+		traceLogsPattern = rotateTraceLogsPattern
+	}
+
+	if ctx.String("rotate-trace-logs-compress") != "" {
+		rotateTraceLogsCompress = ctx.String("rotate-trace-logs-compress")
+		traceLogsCompress = rotateTraceLogsCompress
+	}
+
 	opts.requestSize = requestSize
 	opts.responseSize = responseSize
+	opts.rotateTraceLogsSize = rotateTraceLogsSize
+	opts.rotateTraceLogsFrequency = rotateTraceLogsFrequency
+	opts.rotateTraceLogsPattern = rotateTraceLogsPattern
+	opts.rotateTraceLogsCompress = rotateTraceLogsCompress
 	return
 }
 
+// Write trace log file
+func writeTracelogsFile(log string) {
+	traceLogsFileSize += len(log)
+	if _, err := traceLogsFile.WriteString(log + "\n"); err != nil {
+		traceLogsFile.Close()
+		fatalIf(probe.NewError(err), "Unable to write trace log file `"+defineTraceLogsFile()+"`")
+	}
+}
+
+// Compress trace log file
+func compressTraceLogsFile() {
+	// Create separate fd for asynch compress
+	switch traceLogsCompress {
+	case "gzip":
+		go compressGzip(traceLogsFile, traceLogsFilePath)
+	case "zstd":
+		go compressZstd(traceLogsFile, traceLogsFilePath)
+	default:
+		break
+	}
+}
+
+// gzip
+func compressGzip(asynchTraceLogsFile *os.File, asynchTraceLogsFilePath string) {
+	// Remove raw file
+	defer func() {
+		if err := os.Remove(asynchTraceLogsFilePath); err != nil {
+			// Ignore "no such file or directory". Another log archive already contains this data
+			if errors.Is(err, os.ErrNotExist) {
+				return
+			}
+			fatalIf(probe.NewError(err), "Unable to remove current trace log file `"+asynchTraceLogsFilePath+"`")
+		}
+	}()
+	compressFile, err := os.OpenFile(asynchTraceLogsFilePath+".gz", os.O_CREATE|os.O_RDWR, 0o644)
+	if err != nil {
+		compressFile.Close()
+		asynchTraceLogsFile.Close()
+		fatalIf(probe.NewError(err), "Unable to open trace log file archive for gzip `"+asynchTraceLogsFilePath+"`")
+	}
+	asynchTraceLogsFile.Seek(0, io.SeekStart)
+	writer := gzip.NewWriter(compressFile)
+	_, err = io.Copy(writer, asynchTraceLogsFile)
```

Comment:
All `go` errors in `mc` are called `e` and all `mc` errors related to probe.Error are called `err` - you should make sure to change this everywhere to avoid variable shadowing everywhere.

Read them in other parts of the code.